### DISCUSSION TITLE
Next.js 14 `fetch` requests show `user_agent=undici` in the backend logs

### DISCUSSION THREAD

**Discussion by ADTC** [vercel/next.js#57875] [App Router] [answered]
Although Undici is removed in Next.js 14, the `fetch` requests coming to the backend server from the Next.js instance in Vercel (or a local instance) show `user_agent=undici`.

Is this a misconfiguration in our code, or a bug in Next.js 14?

**Comment by icyJoseph** [comment]
This is coming from Node's side right?

1. Go to: https://public.requestbin.com/r
2. Copy the link of the bin
3. Create a local script `test.js`
```js
async function run() {
  const req = await fetch('<request bin url>');
  console.log(req.status);
}

run();
```
4. `node test.js` with node 18+ (the ones that have native Fetch)
5. Go back to the request bin and see the User Agent
<img width="422" alt="Screenshot 2023-11-01 at 13 26 37" src="https://github.com/vercel/next.js/assets/21013447/2adc7fe3-cb27-4f37-bcef-4249231ca0fb">

The user-agent is `undici`, no Next.js was used at all.

- https://github.com/nodejs/undici/issues/1475

**Reply by ADTC** [reply]
Yeah, I'm also getting the same result. Node version is `20.5.1`.

<img width="400" alt="image" src="https://github.com/vercel/next.js/assets/6047296/2a910bf6-8f85-485a-890c-7d39da5279f9">

So this is an issue with Node?

Actually, I don't understand, if Node uses native fetch since version 18, why is the user-agent undici?

**Reply by ADTC** [reply]
@icyJoseph is this issue related?
* https://github.com/nodejs/undici/issues/2170

**Reply by icyJoseph** [reply, marked as answer]
Because Node's fetch family of APIs are built using undici.

**Reply by ADTC** [reply]
Okay, I think I must have misunderstood what _"drop undici"_ https://github.com/vercel/next.js/pull/56899 in Next.js 14 changelog meant. If I now understand correctly, it means **the polyfill to add Undici in Node 16 environments was removed, because Next.js 14 dropped support for Node 16.**

Since Next.js 14 only supports Node 18 and above, the polyfill Undici is no longer required, and Next.js automatically uses the native built-in Undici of Node. That's why it was removed from Next.js. But Node.js natively uses Undici under the hood.

Is this correct?

**Reply by ADTC** [reply]
@icyJoseph thank you btw. Much appreciated. 🙂

**Reply by icyJoseph** [reply]
You got it right.
